### PR TITLE
feat(segments): add segments duplicate and edit

### PR DESCRIPTION
**Pull request by jingnanl**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Review by zxkane — COMMENTED** [review]
- [x] update e2e test results

File: frontend/src/pages/analytics/segments/AddUserSegment.tsx [hunk 1]
```diff
@@ -16,26 +16,37 @@ import {
...
 import { SegmentProvider } from 'context/SegmentContext';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useParams } from 'react-router-dom';
-import { INIT_SEGMENT_OBJ } from 'ts/const';
+import { DEFAULT_SEGMENT_GROUP_DATA, INIT_SEGMENT_OBJ } from 'ts/const';
 import { defaultStr } from 'ts/utils';
 import SegmentEditor from './components/SegmentEditor';
+import { getSegmentById } from '../../../apis/segments';
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/segments/AddUserSegment.tsx:33]: change to  `apis/segments`

File: frontend/src/pages/analytics/segments/AddUserSegment.tsx [hunk 2]
```diff
@@ -47,14 +58,44 @@ const AddUserSegments: React.FC = () => {
       href: `/analytics/${projectId}/app/${appId}/segments`,
     },
     {
-      text: t('breadCrumb.createSegment'),
+      text:
+        actionType === 'edit'
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/segments/AddUserSegment.tsx:62]: Change to `ternary(condition, a, b)` from `utils.ts`

File: frontend/src/pages/analytics/segments/AddUserSegment.tsx (hunk 1 above)
Comment (jingnanl) [reply frontend/src/pages/analytics/segments/AddUserSegment.tsx:33]: done

File: frontend/src/pages/analytics/segments/AddUserSegment.tsx (hunk 2 above)
Comment (jingnanl) [reply frontend/src/pages/analytics/segments/AddUserSegment.tsx:62]: done

**Review by chenhaiyun — APPROVED** [review]

**Pull request merged**